src/CoreWCF.Primitives/src/CoreWCF/Telemetry/ActionMetadata.cs

```diff
@@ -0,0 +1,17 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace CoreWCF.Telemetry;
+
+internal sealed class ActionMetadata
+{
+    public ActionMetadata(string? contractName, string operationName)
+    {
+        ContractName = contractName;
+        OperationName = operationName;
+    }
+
+    public string? ContractName { get; set; }
+
```

Comment:
Strings are nullable as they are references. We haven't done the work to enable nullability in this code base so no need to specify.